src/System.Private.CoreLib/shared/System/Threading/ExecutionContext.cs

```diff
@@ -367,19 +323,29 @@ internal static void RunForThreadPoolUnsafe<TState>(ExecutionContext executionCo
             CheckThreadPoolAndContextsAreDefault();
             Debug.Assert(executionContext != null && !executionContext.m_isDefault, "ExecutionContext argument is Default.");
 
-            Thread currentThread = Thread.CurrentThread;
             // Restore Non-Default context
-            currentThread.ExecutionContext = executionContext;
-            if (executionContext.HasChangeNotifications)
-            {
-                OnValuesChanged(previousExecutionCtx: null, executionContext);
-            }
+            RestoreChangedContextToThread(Thread.CurrentThread, executionContext, currentContext: null);
```

Comment:
Not sure about this one. All the others are conditional on the context being different; however here we know the context is different so its always run; may be better to keep it inline?

Reply:
As long as it does not have material impact on throughput, I think less stamped generic code is better.

Reply:
I'll move this one back as its different (as its also known the other context is `null` so it does less checking)